frontend/src/app/homepage-offer/offers.service.ts

```diff
@@ -1,32 +1,40 @@
 import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
-
 import 'rxjs/add/operator/map';
 import { Observable } from 'rxjs/Observable';
 
 import { environment } from '../../environments/environment';
+import { ApiOffer, AppOffer } from './offers.model';
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
 import { loadDefaultImage } from './offer.utils';
-import { Offer } from './offers.model';
-
 
 @Injectable()
 export class OffersService {
   private url = `${environment.apiRoot}/offers/`;
 
   constructor (private http: HttpClient) { }
 
-  getOffers(): Observable<Offer[]> {
-    return this.http.get<Offer[]>(this.url)
+  getOffers(): Observable<ApiOffer[]> {
+    return this.http.get<ApiOffer[]>(this.url)
       .map(offers => offers.map(offer => loadDefaultImage(offer)));
   }
 
-  getOffer(id: number): Observable<Offer> {
-    return this.http.get<Offer>(`${this.url}${id}/`)
+  getOffer(id: number): Observable<ApiOffer> {
+    return this.http.get<ApiOffer>(`${this.url}${id}/`)
       .map(offer => loadDefaultImage(offer));
   }
 
-  getJoinViewUrl(offer: Offer): string {
+  getJoinViewUrl(offer: ApiOffer): string {
     return `${environment.djangoRoot}/offers/${offer.slug}/${offer.id}/join`;
   }
 
+  createOffer(offer: AppOffer)  {
+    return this.http.post(`${environment.apiRoot}/offers/`, offer, { withCredentials: true, observe: 'response' })
```

Comment:
please remove '`{ withCredentials: true, observe: 'response'}`, first one is not needed since after migrating to new HttpClient we have an interceptor that adds it automatically, the second one is not needes, since we don't need to check for status. I will explain it more in places where you subscribe to the observables